routing/pathfind.go

```diff
@@ -446,6 +446,48 @@ func edgeWeight(lockedAmt lnwire.MilliSatoshi, fee lnwire.MilliSatoshi,
 	return int64(fee) + timeLockPenalty
 }
 
+// findPathParams wraps the set of parameters passed to findPath.
+type findPathParams struct {
+	// tx can be set to an existing db transaction. If not set, a new
+	// transaction will be started.
+	tx *bolt.Tx
+
+	// graph is the ChannelGraph to be used during path finding.
+	graph *channeldb.ChannelGraph
+
+	// additionalEdges is an optional set of edges that should be
+	// considered during path finding, that is not already found in the
+	// channel graph.
+	additionalEdges map[Vertex][]*channeldb.ChannelEdgePolicy
+
+	// sourceNode is the starting point of the path finding attempt.
+	sourceNode *channeldb.LightningNode
+
+	// targetNode  is the destination of the path finding attempt.
+	targetNode *btcec.PublicKey
+
+	// ignoredNodes is an optional set of nodes that should be ignored if
+	// encountered during path finding.
+	ignoredNodes map[Vertex]struct{}
+
+	// ignoredEdges is an optional set of edges that should be ignored if
+	// encountered during path finding.
+	ignoredEdges map[uint64]struct{}
+
+	// amt is the amount that should be sent to the target over the seeked
+	// path.
+	amt lnwire.MilliSatoshi
+
+	// feeLimit is a maximum fee amount allowed to be used on the path from
+	// the source to the target.
+	feeLimit lnwire.MilliSatoshi
+
+	// bandwidthHints is an optional map from channels to bandwidths that
+	// can be populated if the caller has a better estimate of the current
+	// channel bandwidth than what is found in the graph.
+	bandwidthHints map[uint64]lnwire.MilliSatoshi
+}
+
```

Comment:
This structs is quite a mix of parameters. How about creating two separate structs?

- graphData (`tx`, `graph`, `additionalEdges`, `bandwidthHints`)

- restrictions (`ignoredNodes`, `ignoredEdges`, `feeLimit`) (#1941 is adding a time lock limit to this)

And keep passing in `sourceNode`, `targetNode` and `amt` as direct function arguments. So 5 arguments in total.

Reply:
Done!